csrc/alias_analysis.cpp

```diff
@@ -332,6 +333,40 @@ void AliasFinder::handle(const SliceOp* slice) {
   analysis_.add(out, in, std::move(out_layout));
 }
 
+void AliasFinder::handle(const SqueezeOp* squeeze) {
+  TensorView* in = dynamic_cast<TensorView*>(squeeze->in());
+  if (in == nullptr) {
+    return;
+  }
+  auto* out = squeeze->out()->as<TensorView>();
+
+  // Look at the preferred layout not `in`'s current layout.
+  Layout in_layout = analysis_.preferredLayout(in);
+  if (!ir_utils::computePermutation(
+           in->getMaybeRFactorDomain(), in_layout.allocation_domain)
+           .has_value()) {
+    // Give up when `in`'s allocation domain is not an rfactor permutation.
+    return;
+  }
+
+  std::unordered_map<IterDomain*, IterDomain*> in_rfactor_to_out_root =
+      PairwiseRootDomainMap(in, out).mapBroadcast(true).mapProducerToConsumer();
+
+  Layout out_layout;
+  // Preserve the allocation order of existing dimensions.
+  for (const auto i : c10::irange(in_layout.allocation_domain.size())) {
+    IterDomain* in_allocation_id = in_layout.allocation_domain[i];
+    if (in_rfactor_to_out_root.count(in_allocation_id) == 0) {
+      continue;
+    }
```

Comment:
You could potentially add a check for `isReduction()` here then factor this out as a generic way to preserve allocation domain of the input, in conjunction with #1583 .

Reply:
Great idea -- I'm indeed tired of writing the same pattern over and over again. Will do that in my next PR.